libs/nucs/src/validator.rs

```diff
@@ -81,20 +92,39 @@ impl NucValidator {
         let token_chain = iter::once(token).chain(proofs.iter().copied()).rev();
         Self::validate_proofs(&proofs, &self.root_keys)?;
         Self::validate_token_chain(token_chain, parameters)?;
-        Self::validate_token(token, &proofs, parameters.require_invocation)?;
+        Self::validate_token(token, &proofs, &parameters.token_requirements)?;
         envelope.validate_signatures()?;
         Ok(())
     }
 
     // Validations applied only to the token itself
-    fn validate_token(token: &NucToken, proofs: &[&NucToken], require_invocation: bool) -> ValidationResult {
+    fn validate_token(
+        token: &NucToken,
+        proofs: &[&NucToken],
+        requirements: &TokenTypeRequirements,
+    ) -> ValidationResult {
         match &token.body {
             TokenBody::Delegation(_) => {
-                if require_invocation {
-                    return Err(ValidationError::Validation(ValidationKind::NeedInvocation));
-                }
+                match requirements {
+                    TokenTypeRequirements::Invocation(_) => {
+                        return Err(ValidationError::Validation(ValidationKind::NeedInvocation));
+                    }
+                    TokenTypeRequirements::Delegation(did) => {
+                        Self::validate_condition(&token.audience == did, ValidationKind::InvalidAudience)?
```

Comment:
i would add expected and found audience values in the error, which can help in debugging later

Reply:
This is already the pattern used all over these checks no? e.g. not-before/expired checks (what is the current time?), issuer/audience mismatch (who is the issuer, who is the audience?), root key signature missing (who is the root?), etc.